recipes/default.rb

```diff
@@ -21,31 +21,52 @@
 require "toml"
 
 gpgkey_url = 'https://mackerel.io/assets/files/GPG-KEY-mackerel'
+gpgkey_url_v2 = 'https://mackerel.io/file/cert/GPG-KEY-mackerel-v2'
 package_options = ""
 
+platform_supports_v2_repository = value_for_platform(
+  ['centos', 'redhat'] => { '>= 7.0' => true },
+  'debian' => { '>= 8.0' => true },
+  'ubuntu' => { '>= 16.04' => true },
+  'default' => false,
+)
+
 if platform?('centos') or platform?('redhat') or platform?('amazon')
+  repo_url = "http://yum.mackerel.io/centos/$basearch"
+  yum_key_name = 'RPM-GPG-KEY-mackerel'
+  if platform?('amazon')
+    repo_url = "http://yum.mackerel.io/amznlinux/$releasever/$basearch"
+  elsif platform_supports_v2_repository and node[:kernel][:machine] === 'x86_64'
+    repo_url = "http://yum.mackerel.io/v2/$basearch"
+    gpgkey_url = gpgkey_url_v2
+    yum_key_name = 'RPM-GPG-KEY-mackerel-v2'
+  end
+
   include_recipe 'yum'
   yum_cookbook_ver = Gem::Version.new(run_context.cookbook_collection['yum'].version)
   if yum_cookbook_ver < Gem::Version.new('3.0.0')
-    yum_key "RPM-GPG-KEY-mackerel" do
+    yum_key yum_key_name do
       url gpgkey_url
       action :add
     end
   end
-  repo_url = "http://yum.mackerel.io/centos/$basearch"
-  repo_url = "http://yum.mackerel.io/amznlinux/$releasever/$basearch" if platform?('amazon')
   yum_repository "mackerel" do
     gpgkey gpgkey_url if yum_cookbook_ver >= Gem::Version.new('3.0.0')
     description "mackerel-agent monitoring"
     url repo_url
     action :add
   end
 elsif platform?('debian') or platform?('ubuntu')
-  package_options = '--force-yes -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
+  package_options = '--yes -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold"'
+  repo_url = 'http://apt.mackerel.io/debian/'
+  if platform_supports_v2_repository and node[:kernel][:machine] === 'x86_64'
```

Comment:
When this cookbook is applied to Debian 8 installed v1 repository (by hand or old cookbook), it's replaced to v2 repository.
I think it's acceptable behavior, but should I add some options like `prefer_v1`/`prefer_v2` ?